app/models/concerns/dates_in_past_validator.rb

```diff
@@ -0,0 +1,10 @@
+class DatesInPastValidator < ActiveModel::Validator
+  def validate(record)
+    if record.instance_of?(Era) && record.start_date && record.end_date
+      record.errors.add(:start_date, 'cannot be in the future') if record.start_date > Date.today
+      record.errors.add(:end_date, 'cannot be in the future') if record.end_date > Date.today
+    elsif record.instance_of?(Event) && record.date
+      record.errors.add(:date, 'cannot be in the future') if record.date > Date.today
+    end
+  end
+end
```

Comment:
This is ugggs. If anyone has a more eloquent way of writing this please let me know.

Reply:
Also, this was really cool to build. Made me feel like a legit dev adding things to `models/concerns` lol

Reply:
Niiiiice! I think it's readable and easy to understand what's happening here.

Reply:
here are some sources in case someone else wants to make a custom validator
[rails docs](https://guides.rubyonrails.org/v2.3.11/activerecord_validations_callbacks.html#creating-custom-validation-methods)
[tutorial](https://hackernoon.com/performing-custom-validations-in-rails-an-example-9a373e807144)

Reply:
I agree, I think it's super readable. Nice job on figuring this out